### PR TITLE
Add specs for missing cases

### PR DESCRIPTION
**Pull request by eregon**
Along with my mostly pure-Ruby implementation for TruffleRuby (#660), I discovered a few incompatibilities in TruffleRuby which were not covered by existing specs.

So in this PR, I add specs for previously untested code.
cc @larskanis

### CODE REVIEW

File: spec/ffi/callback_spec.rb [hunk 1]
```diff
@@ -500,27 +507,39 @@ class S8F32S32 < FFI::Struct
...
+
+    # Using a FFI::Function (v2) should be consistent with the direct callback (v1)"
+    v2 = 0xdeadbeef
+    fun = FFI::Function.new(:void, [:uchar]) { |i| v2 = yield(i) }
+    LibTest.testCallbackU8rV(fun, value)
+    expect(v2).to eq(value)
+  end
+
   it ":uchar (0) argument" do
     v = 0xdeadbeef
```
Comment (larskanis) [inline spec/ffi/callback_spec.rb]: You can remove this and the `expect` line below as well, can't you?

Comment (eregon) [reply spec/ffi/callback_spec.rb]: Good catch, indeed.

Comment (eregon) [reply spec/ffi/callback_spec.rb]: This is fixed now (I originally wrote the helper a bit differently, but the latest version seems the simplest).

**Comment by eregon** [comment]
@larskanis Could you merge this? :)

**Comment by eregon** [comment]
Thanks!

**Pull request merged**